wweval/R/ms_fig4.R

```diff
@@ -0,0 +1,519 @@
+      "model", "horizon"
+    ))
+
+  scores_overall <- scores |>
+    data.table::as.data.table() |>
+    scoringutils::summarise_scores(by = c(
+      "forecast_date", "location",
+      "model"
+    )) |>
+    dplyr::mutate(horizon = "overall")
+
+  scores_comb <- dplyr::bind_rows(scores_by_horizon, scores_overall) |>
+    dplyr::filter(
+      horizon %in% horizons_to_show
+    ) |>
+    dplyr::mutate(
+      fig_order = dplyr::case_when(
+        horizon == "nowcast" ~ 1,
+        horizon == "1 wk" ~ 2,
+        horizon == "2 wks" ~ 3,
+        horizon == "3 wks" ~ 4,
+        horizon == "4 wks" ~ 5,
+        horizon == "overall" ~ 6
+      )
+    )
+
+  relative_crps <- scores_comb |>
+    dplyr::select(
+      location, forecast_date, model, horizon,
+      fig_order, crps
+    ) |>
+    dplyr::filter(!is.na(horizon)) |>
+    tidyr::pivot_wider(
+      names_from = model,
+      values_from = crps,
+      id_cols = c(location, forecast_date, horizon, fig_order)
+    ) |>
+    dplyr::mutate(
+      rel_crps = ww / hosp
+    ) |>
+    dplyr::mutate(
+      horizon = forcats::fct_reorder(horizon, fig_order)
+    )
+
+  p <- ggplot(
+    relative_crps,
+    aes(
+      x = as.factor(forecast_date), y = rel_crps, color = horizon,
+      fill = horizon
+    )
+  ) +
+    geom_violin(alpha = 0.3) +
+    geom_hline(aes(yintercept = 1), linetype = "dashed") +
+    theme_bw() +
+    theme(
+      axis.text.x = element_text(
+        size = 8, vjust = 1,
+        hjust = 1, angle = 45
+      ),
+      axis.title.x = element_text(size = 12),
+      axis.title.y = element_text(size = 10),
+      plot.title = element_text(
+        size = 10,
+        vjust = 0.5, hjust = 0.5
+      )
+    ) +
+    ggtitle("CRPS across forecast dates") +
+    xlab("") +
+    ylab("Relative CRPS, lower is better") +
+    scale_y_continuous(trans = "log")
+
+  return(p)
+}
+
+#' Make a figure of the percent of locations with a better forecast with ww
+#' over time
+#'
+#' @param scores A tibble of scores by location, forecast date, date and model,
+#' the ouput of `scoringutils::score()` on samples.
+#' @param eval_hosp_data The retrospective hospital admissions data for each
+#' state and for all dates, used to generate national hospital admissions
+#' @param days_to_show_prev_data An ingeger indicating how many days before the
+#' last forecast date to show the data, default is `14`
+#'
+#' @return a ggplot object containing a stacked bar chart of the percent of
+#' states with imprved forecasts from wastewater by forecast date alongside
+#' daily national hospital admissions
+#' @export
+make_fig4_pct_better_w_ww <- function(scores,
+                                      eval_hosp_data,
+                                      days_to_show_prev_data = 14) {
+  pct_better_w_ww <- scores |>
+    data.table::as.data.table() |>
+    scoringutils::summarise_scores(by = c(
+      "forecast_date", "location",
+      "model"
+    )) |>
+    tibble::as_tibble() |>
```

Comment:
I believe that this is not necessary. `tidyr::pivot_wider` should convert the `data.frame` to a `tibble` internally.

```suggestion
```